### ISSUE TITLE
ESLint v9 Support? (flat config)

### ISSUE THREAD

**Issue by timble-one** [relayjs/eslint-plugin-relay#156]
## Compatibility Problem? - TypeError: Error while loading rule 'relay/no-future-added-value': Rule must be an object with a `create` method

**eslint.config.js**

```JavaScript
export default [
  ...tseslint.config(
    { ignores: ['dist'] },
    {
      files: ['**/*.{ts,tsx}'],
      languageOptions: {
        ecmaVersion: 2020,
        globals: globals.browser,
      },
      extends: [js.configs.recommended, ...tseslint.configs.recommended, relay.configs.recommended],
      rules: {
        ...reactHooks.configs.recommended.rules,
        ...reactRefresh.configs.recommended.rules,
        'react-refresh/only-export-components': [
          'warn',
          { allowConstantExport: true },
        ],
        'react-hooks/exhaustive-deps': 'off',
      },
      plugins: {
        'react-hooks': reactHooks,
        'react-refresh': reactRefresh,
        relay
      },
    },
  ),
]
```

**Error**

```
Oops! Something went wrong! :(

ESLint: 9.20.0

TypeError: Error while loading rule 'relay/no-future-added-value': Rule must be an object with a `create` method
Occurred while linting /Users/timo/projects/flexirent/flexirent-frontend/src/app/pages/Page.tsx
    at createRuleListeners (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:1002:15)
    at /Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:1144:84
    at Array.forEach (<anonymous>)
    at runRules (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:1075:34)
    at #flatVerifyWithoutProcessors (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:2001:31)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:2083:49)
    at Linter._verifyWithFlatConfigArray (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:2172:21)
    at Linter.verify (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:1626:61)
    at Linter.verifyAndFix (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/linter/linter.js:2410:29)
    at verifyText (/Users/timo/projects/flexirent/flexirent-frontend/node_modules/eslint/lib/eslint/eslint.js:327:48)
```

### Workaround
First i tried to disable this rule (`relay/no-future-added-value`) but it looks like all of the rules cause this error.

I also tried to use the compability-library from eslint: https://github.com/eslint/eslintrc#usage  
Doesn't solve the problem neither.

I downgraded eslint to v8.5 and backported `eslint.config.js` to `.eslintrc.cjs`  
But because eslint v8 is end of live this is not really satisfying.

## Import Problem (solved) - TypeError: relay.configs.recommended is not iterable

it looks like this plugin does not yet support the new flat config system introduced in ESLint v9.

If i do this:
```
export default tseslint.config(
  { ignores: ['dist'] },
  {
    extends: [
      js.configs.recommended,
      ...tseslint.configs.recommended,
      ...relay.configs.recommended
    ],
```

I get this:  

`TypeError: relay.configs.recommended is not iterable`

### Solution for Import Problem
This was a stupid mistake of mine.

If i import it like this, this works: `import relay from 'eslint-plugin-relay'`

**Comment by net** [comment]
#157 fixes the errors.

**Comment by MTG2000** [comment]
@net 
It would be great if the above PR fixes this issue.

I haven't been able to migrate to ESLint 9 for a while now because I keep getting the same error.

I updated my config file using the `npx @eslint/migrate-config` script, and my file looks something like:
```js
import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { FlatCompat } from '@eslint/eslintrc'
import js from '@eslint/js'
import typescriptEslint from '@typescript-eslint/eslint-plugin'
import tsParser from '@typescript-eslint/parser'
import relay from 'eslint-plugin-relay'

const __filename = fileURLToPath(import.meta.url)
const __dirname = path.dirname(__filename)
const compat = new FlatCompat({
  baseDirectory: __dirname,
  recommendedConfig: js.configs.recommended,
  allConfig: js.configs.all,
})

export default [
  {
    ignores: [
      '**/dist',
    ],
  },
  ...compat.extends('plugin:relay/recommended'), 
  {
    plugins: {
      '@typescript-eslint': typescriptEslint,
      relay,
    },
    languageOptions: {
      parser: tsParser,
    },
    rules: {
      'relay/generated-flow-types': 'off',
    },
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
  },
]
```
So I think it shouldn't be a case of misconfiguration.

**Comment by net** [comment]
@MTG2000 If you trust me, you can point your eslint-plugin-relay dependency at my GitHub fork until the maintainers merge the PR, and follow the example in the PR description.
